libbpf-tools/memleak.bpf.c

```diff
@@ -0,0 +1,374 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 Jackie Dinh
+//
+// Based on memleak(8) from BCC by Sasha Goldshtein
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include "memleak.h"
+
+char LICENSE[] SEC("license") = "GPL";
+
+const volatile size_t min_size = 0;
+const volatile size_t max_size = 1UL << 32;
+const volatile int sample_rate = 1;
+const volatile bool trace_all = false;
+const volatile long page_size = 4096;
+const volatile int stack_flags = 0;
+const volatile bool wa_missing_free = false;
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_HASH_ENTRY_NUM);
+	__type(key, __u64);
+	__type(value, __u64);
+} sizes SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_HASH_ENTRY_NUM);
+	__type(key, __u64);
+	__type(value, struct alloc_info_t);
+} allocs SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_HASH_ENTRY_NUM);
+	__type(key, __u64);
+	__type(value, __u64);
+} memptrs SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_STACK_TRACE);
+	__uint(key_size, sizeof(__u32));
+} stack_traces SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_HASH_ENTRY_NUM);
+	__type(key, __u64);
+	__type(value, struct combined_alloc_info_t);
+} combined_allocs SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, u32);
+	__type(value, u64);
+} alloc_counters SEC(".maps");
+
+static __always_inline void update_statistics_add(__u64 stack_id, __u64 sz) {
```

Comment:
Drop all `__always_inline` attributes.

Reply:
Fix in commit [2beb289e](https://github.com/iovisor/bcc/pull/3828/commits/2beb289e98d651bddc2127299c5627b5d6149fce).